sljit_src/sljitProtExecAllocator.c

```diff
@@ -204,6 +203,36 @@ static SLJIT_INLINE struct chunk_header* alloc_chunk(sljit_uw size)
 	retval->fd = fd;
 	return retval;
 }
+#else
+static SLJIT_INLINE struct chunk_header* alloc_chunk(sljit_uw size)
+{
+	struct chunk_header *retval;
+	void *maprx;
+
+	retval = (struct chunk_header *)mmap(NULL, size,
+			PROT_MPROTECT(PROT_EXEC|PROT_WRITE|PROT_READ),
+			MAP_ANON, -1, 0);
+
+	if (retval == MAP_FAILED)
+		return NULL;
+
+	maprx = mremap(retval, size, NULL, size, MAP_REMAPDUP);
```

Comment:
Is this code NetBSD spcefic, or any system with MAP_REMAPDUP can be supported (at least in theory)?

Reply:
MAP_REMAPDUP is NetBSD specific and available since version 8, hopefully other systems could reuse this code if they implement MAP_REMAPDUP or a equivalent but none is available as of now, hence why the whole code segment is ifdef to __NetBSD__

indeed, even in NetBSD, version 7 (soon to be EOL) or older don't have MAP_REMAPDUP and will use the original file based code instead (tested with 7.2 in i386)

Reply:
In the original code there was a guard around `secure_getenv`.  Is this not needed in older NetBSD systems anymore?

Reply:
nothing has changed regarding portability of that function, which is why that old code guard is still needed and will be used in NetBSD 7 or older.

a slightly more sophisticated version of it will be needed for OpenBSD or HardenedBSD but that change is orthogonal to this one